internal/provider/resource_ably_rule_kinesis.go

```diff
@@ -0,0 +1,433 @@
+package ably_control
+
+import (
+	"context"
+
+	ably_control_go "github.com/ably/ably-control-go"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	tfsdk_provider "github.com/hashicorp/terraform-plugin-framework/provider"
+	tfsdk_resource "github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+type resourceRuleKinesisType struct{}
+
+// Get Rule Resource schema
+func (r resourceRuleKinesisType) GetSchema(_ context.Context) (tfsdk.Schema, diag.Diagnostics) {
+	return tfsdk.Schema{
+		Attributes: map[string]tfsdk.Attribute{
+			"id": {
+				Type:        types.StringType,
+				Computed:    true,
+				Description: "The rule ID.",
+				PlanModifiers: []tfsdk.AttributePlanModifier{
+					tfsdk_resource.UseStateForUnknown(),
+				},
+			},
+			"app_id": {
+				Type:        types.StringType,
+				Required:    true,
+				Description: "The Ably application ID.",
+			},
+			"status": {
+				Type:        types.StringType,
+				Optional:    true,
+				Description: "The status of the rule. Rules can be enabled or disabled.",
+			},
+			"rule_type": {
+				Type:        types.StringType,
+				Required:    true,
+				Description: "The type of rule. See the documentation for further information.",
+			},
+			"source": {
+				Required:    true,
+				Description: "object (rule_source)",
+				Attributes: tfsdk.SingleNestedAttributes(map[string]tfsdk.Attribute{
+					"channel_filter": {
+						Type:     types.StringType,
+						Required: true,
+					},
+					"type": {
+						Type:     types.StringType,
+						Required: true,
+					},
+				}),
+			},
+			"aws_authentication": {
+				Optional:    true,
+				Description: "object (rule_source)",
+				Attributes: tfsdk.SingleNestedAttributes(map[string]tfsdk.Attribute{
+					"mode": {
+						Type:     types.StringType,
+						Optional: true,
+						PlanModifiers: []tfsdk.AttributePlanModifier{
+							tfsdk_resource.RequiresReplace(),
+						},
+					},
+					"role_arn": {
+						Type:     types.StringType,
+						Optional: true,
+					},
+					"access_key_id": {
+						Type:      types.StringType,
+						Optional:  true,
+						Sensitive: true,
+					},
+					"secret_access_key": {
```

Comment:
Can you really not edit the key without replacing it?

Reply:
You're right. It's not required so I've removed it.